pkg/config/stepmeta.go

```diff
@@ -301,7 +304,9 @@ func (m *StepData) GetContextDefaults(stepName string) (io.ReadCloser, error) {
 		root["sidecarEnvVars"] = EnvVarsAsMap(m.Spec.Sidecars[0].EnvVars)
 		root["sidecarImage"] = m.Spec.Sidecars[0].Image
 		root["sidecarName"] = m.Spec.Sidecars[0].Name
-		root["sidecarPullImage"] = m.Spec.Sidecars[0].ImagePullPolicy != "Never"
+		if m.Spec.Sidecars[0].ImagePullPolicy != "" {
```

Comment:
[side catch] Bad to have this twice. Basically the properties for the sidecar and the "normal" image are the same.  Here we repeat the same fix a second time. All the other parameters are not handled with `putXXXIfNotEmpty`. Is there a reason for not doing this here, too (which in fact is also applying the same thing twice.

Reply:
Thanks for finding this, I will improve. We definitely also should use `putXXXIfNotEmpty` for the sidecar container options.

Reply:
Yes, this was an oversight.